// react-dom/lib/ReactInjection.js
define('node_modules/react-dom/lib/ReactInjection', function(require, exports, module) {

  /**
   * Copyright 2013-present, Facebook, Inc.
   * All rights reserved.
   *
   * This source code is licensed under the BSD-style license found in the
   * LICENSE file in the root directory of this source tree. An additional grant
   * of patent rights can be found in the PATENTS file in the same directory.
   *
   */
  
  'use strict';
  
  var DOMProperty = require('node_modules/react-dom/lib/DOMProperty');
  var EventPluginHub = require('node_modules/react-dom/lib/EventPluginHub');
  var EventPluginUtils = require('node_modules/react-dom/lib/EventPluginUtils');
  var ReactComponentEnvironment = require('node_modules/react-dom/lib/ReactComponentEnvironment');
  var ReactEmptyComponent = require('node_modules/react-dom/lib/ReactEmptyComponent');
  var ReactBrowserEventEmitter = require('node_modules/react-dom/lib/ReactBrowserEventEmitter');
  var ReactHostComponent = require('node_modules/react-dom/lib/ReactHostComponent');
  var ReactUpdates = require('node_modules/react-dom/lib/ReactUpdates');
  
  var ReactInjection = {
    Component: ReactComponentEnvironment.injection,
    DOMProperty: DOMProperty.injection,
    EmptyComponent: ReactEmptyComponent.injection,
    EventPluginHub: EventPluginHub.injection,
    EventPluginUtils: EventPluginUtils.injection,
    EventEmitter: ReactBrowserEventEmitter.injection,
    HostComponent: ReactHostComponent.injection,
    Updates: ReactUpdates.injection
  };
  
  module.exports = ReactInjection;

});
