// draft-js/lib/DraftEditorEditHandler.js
define('node_modules/draft-js/lib/DraftEditorEditHandler', function(require, exports, module) {

  /**
   * Copyright (c) 2013-present, Facebook, Inc.
   * All rights reserved.
   *
   * This source code is licensed under the BSD-style license found in the
   * LICENSE file in the root directory of this source tree. An additional grant
   * of patent rights can be found in the PATENTS file in the same directory.
   *
   * @providesModule DraftEditorEditHandler
   * 
   */
  
  'use strict';
  
  var onBeforeInput = require('node_modules/draft-js/lib/editOnBeforeInput');
  var onBlur = require('node_modules/draft-js/lib/editOnBlur');
  var onCompositionStart = require('node_modules/draft-js/lib/editOnCompositionStart');
  var onCopy = require('node_modules/draft-js/lib/editOnCopy');
  var onCut = require('node_modules/draft-js/lib/editOnCut');
  var onDragOver = require('node_modules/draft-js/lib/editOnDragOver');
  var onDragStart = require('node_modules/draft-js/lib/editOnDragStart');
  var onFocus = require('node_modules/draft-js/lib/editOnFocus');
  var onInput = require('node_modules/draft-js/lib/editOnInput');
  var onKeyDown = require('node_modules/draft-js/lib/editOnKeyDown');
  var onPaste = require('node_modules/draft-js/lib/editOnPaste');
  var onSelect = require('node_modules/draft-js/lib/editOnSelect');
  
  var DraftEditorEditHandler = {
    onBeforeInput: onBeforeInput,
    onBlur: onBlur,
    onCompositionStart: onCompositionStart,
    onCopy: onCopy,
    onCut: onCut,
    onDragOver: onDragOver,
    onDragStart: onDragStart,
    onFocus: onFocus,
    onInput: onInput,
    onKeyDown: onKeyDown,
    onPaste: onPaste,
    onSelect: onSelect
  };
  
  module.exports = DraftEditorEditHandler;

});
