// draft-js/lib/Draft.js
define('node_modules/draft-js/lib/Draft', function(require, exports, module) {

  /**
   * Copyright (c) 2013-present, Facebook, Inc.
   * All rights reserved.
   *
   * This source code is licensed under the BSD-style license found in the
   * LICENSE file in the root directory of this source tree. An additional grant
   * of patent rights can be found in the PATENTS file in the same directory.
   *
   * @providesModule Draft
   */
  
  'use strict';
  
  var AtomicBlockUtils = require('node_modules/draft-js/lib/AtomicBlockUtils');
  var BlockMapBuilder = require('node_modules/draft-js/lib/BlockMapBuilder');
  var CharacterMetadata = require('node_modules/draft-js/lib/CharacterMetadata');
  var CompositeDraftDecorator = require('node_modules/draft-js/lib/CompositeDraftDecorator');
  var ContentBlock = require('node_modules/draft-js/lib/ContentBlock');
  var ContentState = require('node_modules/draft-js/lib/ContentState');
  var DefaultDraftBlockRenderMap = require('node_modules/draft-js/lib/DefaultDraftBlockRenderMap');
  var DefaultDraftInlineStyle = require('node_modules/draft-js/lib/DefaultDraftInlineStyle');
  var DraftEditor = require('node_modules/draft-js/lib/DraftEditor.react');
  var DraftEditorBlock = require('node_modules/draft-js/lib/DraftEditorBlock.react');
  var DraftEntity = require('node_modules/draft-js/lib/DraftEntity');
  var DraftModifier = require('node_modules/draft-js/lib/DraftModifier');
  var DraftEntityInstance = require('node_modules/draft-js/lib/DraftEntityInstance');
  var EditorState = require('node_modules/draft-js/lib/EditorState');
  var KeyBindingUtil = require('node_modules/draft-js/lib/KeyBindingUtil');
  var RichTextEditorUtil = require('node_modules/draft-js/lib/RichTextEditorUtil');
  var SelectionState = require('node_modules/draft-js/lib/SelectionState');
  
  var convertFromDraftStateToRaw = require('node_modules/draft-js/lib/convertFromDraftStateToRaw');
  var convertFromHTMLToContentBlocks = require('node_modules/draft-js/lib/convertFromHTMLToContentBlocks');
  var convertFromRawToDraftState = require('node_modules/draft-js/lib/convertFromRawToDraftState');
  var generateRandomKey = require('node_modules/draft-js/lib/generateRandomKey');
  var getDefaultKeyBinding = require('node_modules/draft-js/lib/getDefaultKeyBinding');
  var getVisibleSelectionRect = require('node_modules/draft-js/lib/getVisibleSelectionRect');
  
  var DraftPublic = {
    Editor: DraftEditor,
    EditorBlock: DraftEditorBlock,
    EditorState: EditorState,
  
    CompositeDecorator: CompositeDraftDecorator,
    Entity: DraftEntity,
    EntityInstance: DraftEntityInstance,
  
    BlockMapBuilder: BlockMapBuilder,
    CharacterMetadata: CharacterMetadata,
    ContentBlock: ContentBlock,
    ContentState: ContentState,
    SelectionState: SelectionState,
  
    AtomicBlockUtils: AtomicBlockUtils,
    KeyBindingUtil: KeyBindingUtil,
    Modifier: DraftModifier,
    RichUtils: RichTextEditorUtil,
  
    DefaultDraftBlockRenderMap: DefaultDraftBlockRenderMap,
    DefaultDraftInlineStyle: DefaultDraftInlineStyle,
  
    convertFromHTML: convertFromHTMLToContentBlocks,
    convertFromRaw: convertFromRawToDraftState,
    convertToRaw: convertFromDraftStateToRaw,
    genKey: generateRandomKey,
    getDefaultKeyBinding: getDefaultKeyBinding,
    getVisibleSelectionRect: getVisibleSelectionRect
  };
  
  module.exports = DraftPublic;

});
