// rc-trigger/lib/LazyRenderBox.js
define('node_modules/rc-trigger/lib/LazyRenderBox', function(require, exports, module) {

  'use strict';
  
  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  
  var _objectWithoutProperties2 = require('node_modules/babel-runtime/helpers/objectWithoutProperties');
  
  var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);
  
  var _react = require('node_modules/react/react');
  
  var _react2 = _interopRequireDefault(_react);
  
  function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
  
  var LazyRenderBox = _react2["default"].createClass({
    displayName: 'LazyRenderBox',
  
    propTypes: {
      children: _react.PropTypes.any,
      className: _react.PropTypes.string,
      visible: _react.PropTypes.bool,
      hiddenClassName: _react.PropTypes.string
    },
    shouldComponentUpdate: function shouldComponentUpdate(nextProps) {
      return nextProps.hiddenClassName || nextProps.visible;
    },
    render: function render() {
      var _props = this.props,
          hiddenClassName = _props.hiddenClassName,
          visible = _props.visible,
          props = (0, _objectWithoutProperties3["default"])(_props, ['hiddenClassName', 'visible']);
  
  
      if (hiddenClassName || _react2["default"].Children.count(props.children) > 1) {
        if (!visible && hiddenClassName) {
          props.className += ' ' + hiddenClassName;
        }
        return _react2["default"].createElement('div', props);
      }
  
      return _react2["default"].Children.only(props.children);
    }
  });
  
  exports["default"] = LazyRenderBox;
  module.exports = exports['default'];

});
